src/plugins/intel_npu/src/backend/include/zero_adapter.hpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2018-2024 Intel Corporation
```

Comment:
I still think this should also go in compiler_adapter directory. Is there a good reason to have an interface only in compiler_adapter and an implementation in the backend? This class is doing a lot of graph manipulation, is only used during compilation and not at all during inference execution, thus I don't think it should be located inside the "backend".

Reply:
Actually I see we already have another identical header in compiler_adapter. Is this a leftover?

Reply:
yep